### PR TITLE
Integer unification part 2 : remove most occurrences of Fixnum/Bignum

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: src/main/ruby/core/type.rb [hunk 1]
```diff
@@ -255,6 +250,14 @@ def self.conversion_mismatch(val, cls, meth, res)
       raise TypeError, "can't convert #{val.class} to #{cls} (#{val.class}##{meth} gives #{res.class})"
     end
 
+    def self.fits_into_int?(val)
+      Integer === val && Truffle.invoke_primitive(:fixnum_fits_into_int, val)
```
Comment (nirvdrum) [inline src/main/ruby/core/type.rb]: This primitive should probably be renamed.

File: src/main/ruby/core/type.rb [hunk 2]
```diff
@@ -255,6 +250,14 @@ def self.conversion_mismatch(val, cls, meth, res)
...
+    end
+
+    def self.fits_into_long?(val)
+      Integer === val && Truffle.invoke_primitive(:fixnum_fits_into_long, val)
```
Comment (nirvdrum) [inline src/main/ruby/core/type.rb]: This primitive should probably be renamed.

File: src/main/java/org/truffleruby/core/numeric/IntegerNodes.java [hunk 3]
```diff
@@ -1579,6 +1579,30 @@ public boolean fitsIntoULongBignum(DynamicObject b) {
 
     }
 
+    @Primitive(name = "fixnum_lower")
```
Comment (nirvdrum) [inline src/main/java/org/truffleruby/core/numeric/IntegerNodes.java]: I think we should move away from the "fixnum" name and make a clean break.

File: src/main/ruby/core/marshal.rb [hunk 4]
```diff
@@ -298,18 +298,12 @@ class String
   end
 end
 
-class Fixnum
```
Comment (nirvdrum) [inline src/main/ruby/core/marshal.rb]: Does MRI retain the old classes so old data can be unmarshalled (e.g., serialized to disk or a DB)? If so, we should retain that as well to make migrating to TruffleRuby easier.

File: src/main/ruby/core/type.rb (hunk 1 above)
Comment (eregon) [reply src/main/ruby/core/type.rb]: You mean `fixnum_fits_into_int`? Could do.

File: src/main/ruby/core/marshal.rb (hunk 4 above)
Comment (eregon) [reply src/main/ruby/core/marshal.rb]: `serialize_integer` handles both Fixnum and Bignum, so this is functionally equivalent and produces the same bytes as before.

File: src/main/ruby/core/type.rb (hunk 1 above)
Comment (nirvdrum) [reply src/main/ruby/core/type.rb]: Yes. No need to do it in this PR, but something to note.

Comment (eregon) [reply src/main/ruby/core/type.rb]: I'll merge this and do it in another PR then.

Comment (eregon) [reply src/main/ruby/core/type.rb]: #1352

**Pull request merged**